src/datasets/config.py

```diff
@@ -168,6 +168,9 @@
 # https://github.com/apache/arrow/blob/master/docs/source/cpp/arrays.rst#size-limitations-and-recommendations)
 DEFAULT_MAX_BATCH_SIZE = 10_000
 
+# Number of examples prefetched in `Dataset.__iter__`
+DEFAULT_ITER_BATCH_SIZE = 10
```

Comment:
Prefetching usually means bring to memory, which is not 100% the case here no ? We just prefetch the Arrow data, not the python objects. Since this is very specific to the optimized function I think we can use a more specific name, maybe `ARROW_READER_BATCH_SIZE_IN_DATASET_ITER`